ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -553,8 +563,18 @@ protected Boolean findNextCompactionAndExecute(boolean computeStats) throws Inte
         compactionTxn.wasSuccessful();
       } catch (Throwable e) {
         LOG.error("Caught exception while trying to compact " + ci +
-            ".  Marking failed to avoid repeated failures", e);
+            ". Deleting the result directories created by the compactor "+ resultDir +
+            " Marking failed to avoid repeated failures", e);
         markFailed(ci, e);
+        // remove the new directories created by compaction
+        if (resultDir != null) {
```

Comment:
FS operations should be wrapped with doAs:
 if (runJobAsSelf(ci.runAs)) {
        CLEANUP()
      } else {
        UserGroupInformation ugi = UserGroupInformation.createProxyUser(ci.runAs,UserGroupInformation.getLoginUser());
        ugi.doAs((PrivilegedExceptionAction<Object>) () -> CLEANUP() });